### ISSUE TITLE
Need a way to configure exit codes

### ISSUE THREAD

**Issue by andrew-glenn** [aws-cloudformation/cfn-lint#1586]
https://github.com/aws-cloudformation/cfn-python-lint/blob/0ad76abe7dbd33dc404cd1cedd85c9643822d347/src/cfnlint/core.py#L55-L66

I'm not finding a way to non-zero exit on Errors only. Currently, based on the logic above, an informational message will trigger a non-zero exit code (8). Working around this in CICD pipelines is frustrating. A way to say `--non-zero-exit-code ERROR` would be splendid.

**Comment by PatMyron** [comment]
Thought about the error codes quite a bit. I'd love for informational level rules to be included in the output without affecting the error code, potentially even as the default behavior

> I'm not finding a way to non-zero exit on Errors only.
```
level  code
          0
E         2
W         4
I         8
EW        6
EI       10
WI       12
EWI      14
```

```bash
echo $? % 4 | bc    # exit code modulo 4 should filter warnings and informational
```

**Comment by andrew-glenn** [comment]
In AWS-Native services (ala CodeBuild, and CodePIpeline that leverages CodeBuild), the shell environment is by-default configured to exit on non-zero status codes. 

There are things that can be done to mitigate cfn-lint's behavior, such as `set +e`, trapping and examining the exit code, and things of that nature. Candidly though, it shouldn't be necessary. 

The current behavior puts me in a position where I can't trust the binary notion that non-zero == failure, and instead need to inspect the exit codes (via source-code, subject to change) to determine the path forward. 

At _minimum_ (5pm Friday, before Vacation minimum), I'd expect to see this documented in --help and the README.

**Comment by andrew-glenn** [comment]
@PatMyron - the documentation seems to indicate that's possible. I'm not finding that to be the case, however.

```
❯ cfn-lint --ignore-checks W templates/*
usage:
Basic: cfn-lint test.yaml
Ignore a rule: cfn-lint -i E3012 -- test.yaml
Configure a rule: cfn-lint -x E3012:strict=false -t test.yaml
Lint all yaml files in a folder: cfn-lint dir/**/*.yaml

CloudFormation Linter

(...)
```

```
❯ cfn-lint --version
cfn-lint 0.33.0
```

**Comment by PatMyron** [comment]
might just be the option order? I think `--template` needs to be explicit if not the first option
try one of these:
```
cfn-lint templates/* --ignore-checks W
cfn-lint --ignore-checks W --template templates/*
```

**Comment by andrew-glenn** [comment]
Closing the gap - both of those suggestions work.

**Comment by kddejong** [comment]
I think the only trick here is if someone wants to see the messages but change the exit code.  I'm glad the `--ignore-checks W` works in this case.

**Comment by aihaddad** [comment]
as @kddejong this is exactly the use case. Legacy code base that we're fixing gradually. I want to be able to see the warning so that I can go and fix them bit by bit, but I don't want it to exit with non-zero so that it breaks Jenkins. Just the way ESLint, for example, works

**Comment by kddejong** [comment]
Another way to write @PatMyron's example
```bash
cfn-lint --include-checks I -- template.yaml || (($? % 4 ? 0 : 1))
```
if exit code is `2` from cfn-lint this will return `1`
if exit code is `4` from cfn-lint this will return `0`
if exit code is `6` from cfn-lint this will return `1`
if exit code is `8` from cfn-lint this will return `0`
if exit code is `12` from cfn-lint this will return `0`
if exit code is `14` from cfn-lint this will return `1`

**Comment by kddejong** [comment]
Given the amount of issues related to this I believe bringing some of the workaround mechanisms into the cfn-lint code base make sense.

**Comment by kddejong** [comment]
created #2346 to help move this logic into cfn-lint

**Comment by Alex-Dettinger** [comment]
@kddejong #2346 is not related to this error. Is a fix for this actively being worked on?

I've seen a number of reported issues that I am also having where github actions treats warnings from cfn-lint as an error and fails since version v0.66.0. The only viable workaround is to disable warnings or the specific warning you have trouble with, which defeats the purpose of warnings. **Suppressing warnings is not a fix to this problem**. 

https://github.com/aws-cloudformation/cfn-lint/issues/2297
https://github.com/aws-cloudformation/cfn-lint/issues/897

**Comment by kddejong** [comment]
Wrong #.  Updated now.  The workarounds above will work for you and release v0.90.0 has a new way to configure exit codes. 

You can look at #2460 to see the readme updates on how to use it.
